llvm/lib/Target/AIE/AIEPostSelectOptimize.cpp

```diff
@@ -214,11 +214,23 @@ std::map<int, Register> collectSubregsChain(const MachineInstr &MI) {
   // E.g. %5:vec512 = INSERT_SUBREG %0, %1, %subreg.sub_256_hi
   //      %6:vec512 = INSERT_SUBREG %5, %2, %subreg.sub_256_lo
   std::function<void(const MachineInstr &)> Impl = [&](const MachineInstr &MI) {
+    if (MI.getOpcode() == TargetOpcode::REG_SEQUENCE) {
```

Comment:
Hi, this implementation looks nice and clean, however this map (was here before) looks inverted. Every time we access it we reverse the results. Is it possible to use the same order as in REG_SEQUENCE? Is there any case were the same virtual subreg appears twice in a REQ_SEQUENCE / INSERT_SUBREG?

Reply:
The reason it's ordered this way is exactly for the reason you mention: it's theoretically possible to have the same subreb multiple times in the sequence. In that case, I want to keep only the first instance I met when iterating bottom-up. I have a test for this particular case. I haven't seen it in real world examples, but you never know 🙈